src/services/player.rs

```diff
@@ -6,30 +6,43 @@ use super::packet::{
     PlayerInfo, PlayerPositionAndLook, SpawnPlayer,
 };
 use std::collections::HashMap;
-use std::convert::TryInto;
-use std::sync::mpsc::Receiver;
+
+use std::sync::mpsc::{Receiver, Sender};
 use uuid::Uuid;
 
-pub fn start<M: Messenger + Clone>(receiver: Receiver<PlayerStateOperations>, messenger: M) {
+pub fn start<M: Messenger + Clone>(
+    receiver: Receiver<PlayerStateOperations>,
+    _sender: Sender<PlayerStateOperations>,
+    messenger: M,
+) {
     let mut players = HashMap::<Uuid, Player>::new();
     let mut entity_conn_ids = HashMap::<i32, Uuid>::new();
+    let mut entity_id = 0;
 
     while let Ok(msg) = receiver.recv() {
-        handle_message(msg, &mut players, &mut entity_conn_ids, messenger.clone())
+        handle_message(
+            msg,
+            &mut players,
+            &mut entity_conn_ids,
+            &mut entity_id,
+            messenger.clone(),
+        )
     }
 }
 
 fn handle_message<M: Messenger>(
     msg: PlayerStateOperations,
     players: &mut HashMap<Uuid, Player>,
     entity_conn_ids: &mut HashMap<i32, Uuid>,
+    entity_id: &mut i32,
     messenger: M,
 ) {
     match msg {
         PlayerStateOperations::New(msg) => {
             let mut player = msg.player;
             if player.entity_id == 0 {
-                player.entity_id = players.len().try_into().expect("too many players");
+                player.entity_id = *entity_id;
```

Comment:
Since we started deleting players when they disconnect we can't use length of the player map as an id anymore